controllers/numaresourcesoperator_controller.go

```diff
@@ -305,6 +311,53 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	return daemonSetsNName, nil
 }
 
+func (r *NUMAResourcesOperatorReconciler) deleteUnusedDaemonSets(ctx context.Context, instance *nropv1alpha1.NUMAResourcesOperator, mcps []*machineconfigv1.MachineConfigPool) error {
+	klog.Info("Delete DS start")
+	var daemonSetList appsv1.DaemonSetList
+	if err := r.List(ctx, &daemonSetList, &client.ListOptions{Namespace: instance.Namespace}); err != nil {
+		klog.Infof("error while getting DS list: %v", err)
+		return err
+	}
+
+	// Generate the names of the DaemonSets that should be running,
+	// one for each MachineConfigPool
+	expectedDaemonSetNames := map[string]struct{}{}
+	for _, mcp := range mcps {
+		generatedDSName := rtestate.GetComponentName(instance.Name, mcp.Name)
+		expectedDaemonSetNames[generatedDSName] = struct{}{}
+	}
+
+	var err error = nil
+	for _, ds := range daemonSetList.Items {
+		if _, ok := expectedDaemonSetNames[ds.Name]; !ok {
+			var owned bool
+			owned, err = r.isOwnedBy(ds.GetObjectMeta(), instance)
+			if err != nil {
+				klog.Infof("error handling [%s]: %v", ds.Name, err)
+				continue
+			}
+			if owned {
+				if err = r.Client.Delete(ctx, &ds); err != nil {
+					klog.Infof("error while deleting [%s]: %v", ds.Name, err)
+					continue
+				}
+				klog.Infof("DS[%s] deleted", ds.Name)
+			}
+		}
+	}
+	return err
+}
+
+func (r *NUMAResourcesOperatorReconciler) isOwnedBy(element metav1.Object, owner metav1.Object) (bool, error) {
```

Comment:
I think you can settle for 
``` return element.GetOwnerReferences()  == owner.GetUID()```

Reply:
If this works, I like it much much more - and would probably make the `ownerreferences` borrowed code no longer necessary.

Reply:
I think more correct is `element.GetOwnerReferences().GetUID() == owner.GetUID()`